### PR TITLE
Create reusable build details component and minor UI enhancements

### PR DESCRIPTION
**Pull request by tombui99**
https://github.com/SSWConsulting/SSW.CodeAuditor/issues/669

- Create a reusable component `BuildDetailSlot` that contains:
    - Breadcrumbs
    - CardSummary
    - DetailsCard
    - Tabs
    - Build version 
- Remove all unused and duplicated components
- Minor UI fix - reposition 'Build version'

### CODE REVIEW

File: ui/src/containers/ArtilleryReport.svelte
Comment (zacharykeeping) [inline ui/src/containers/ArtilleryReport.svelte:1]: Looks like "showArtilleryThreshold" is no longer used here - can we remove it?

File: ui/src/containers/ArtilleryReport.svelte [hunk 1]
```diff
@@ -84,60 +81,53 @@
...
-                class="bg-gray-300 hover:bg-gray-400 textdark font-bold py-1 px-1
-              rounded-lg inline-flex items-center">
-                <Icon cssClass="">
-                  <path
-                    d="M4 16v1a3 3 0 003 3h10a3 3 0 003-3v-1m-4-4l-4 4m0 0l-4-4m4 4V4" />
-                </Icon>
-              </button>
+        <BuildDetailsSlot
+          {data}
+          componentType="artillery"
```
Comment (zacharykeeping) [inline ui/src/containers/ArtilleryReport.svelte:86]: Can we have these display with uppercase casing to match the other breadcrumbs parts? (i.e. "Artillery", "URL", "Lighthouse" etc)

Screenshot 2023-09-28 at 3 09 09 pm

**Review by zacharykeeping — CHANGES_REQUESTED** [review]

File: ui/src/containers/ArtilleryReport.svelte
Comment (tombui99) [reply ui/src/containers/ArtilleryReport.svelte:1]: Done - Removed unused Artillery threshold modal

File: ui/src/containers/ArtilleryReport.svelte (hunk 1 above)
Comment (tombui99) [reply ui/src/containers/ArtilleryReport.svelte]: Done - display uppercase in breadcrumb for consistency

image

**Review by zacharykeeping — APPROVED** [review]

**Comment by zacharykeeping** [comment]
LGTM!

I'd noticed there was a lot of duplicated code here so it's good to see this cleaned up. 🙂

**Pull request merged**